engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/gateway/InclusiveGatewayTest.java

```diff
@@ -676,6 +677,246 @@ public void shouldEvaluateConditionWithoutDefaultFlow() {
+            .parallelGateway("fork")
+            .sequenceFlowId("joinFlow1")
+            .inclusiveGateway("join")
+            .moveToNode("fork")
+            .serviceTask("task", b -> b.zeebeJobType("type"))
+            .sequenceFlowId("joinFlow2")
+            .connectTo("join")
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    // complete the job
+    ENGINE.job().ofInstance(processInstanceKey).withType("type").complete();
+
+    // then
+    final List<Record<ProcessInstanceRecordValue>> events =
+        RecordingExporter.processInstanceRecords()
+            .limit(
+                r ->
+                    "join".equals(r.getValue().getElementId())
+                        && ProcessInstanceIntent.ELEMENT_COMPLETED == r.getIntent())
+            .collect(Collectors.toList());
+
+    assertThat(events)
+        .extracting(e -> e.getValue().getElementId(), Record::getIntent)
+        .containsSubsequence(
+            tuple("joinFlow1", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN),
+            tuple("joinFlow2", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN))
+        .containsOnlyOnce(tuple("join", ProcessInstanceIntent.ELEMENT_ACTIVATING));
+  }
+
+  @Test
+  public void shouldTriggerGatewayWhenAllSatisfiedBranchesAreActivatedWithBoundaryEvent() {
+    // given
+    final var process =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .parallelGateway("fork")
+            .sequenceFlowId("joinFlow1")
+            .inclusiveGateway("join")
+            .moveToNode("fork")
+            .serviceTask("task", b -> b.zeebeJobType("type"))
+            .boundaryEvent("event", b -> b.timerWithDuration("PT5S"))
+            .sequenceFlowId("joinFlow2")
+            .connectTo("join")
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+    ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE.increaseTime(Duration.ofSeconds(10));
+
+    // then
+    final List<Record<ProcessInstanceRecordValue>> events =
+        RecordingExporter.processInstanceRecords()
+            .limit(
+                r ->
+                    "join".equals(r.getValue().getElementId())
+                        && ProcessInstanceIntent.ELEMENT_COMPLETED == r.getIntent())
+            .collect(Collectors.toList());
+
+    assertThat(events)
+        .extracting(e -> e.getValue().getElementId(), Record::getIntent)
+        .containsSubsequence(
+            tuple("joinFlow1", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN),
+            tuple("joinFlow2", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN))
+        .containsOnlyOnce(tuple("join", ProcessInstanceIntent.ELEMENT_ACTIVATING));
+  }
+
+  @Test
+  public void shouldMergeAndSplitInOneGateway() {
```

Comment:
:+1: Great test coverage on the new converging inclusive gateway behavior!

:x: However, I think we should add a test case when one gateway splits and merges, but only one, or a subset of the incoming sequence flows are activated.